src/Draco.Compiler.Tests/Semantics/CodeCompletionTests.cs

```diff
@@ -0,0 +1,402 @@
+using System.Collections.Immutable;
+using Draco.Compiler.Api;
+using Draco.Compiler.Api.CodeCompletion;
+using Draco.Compiler.Api.Semantics;
+using Draco.Compiler.Api.Syntax;
+
+namespace Draco.Compiler.Tests.Semantics;
+
+public sealed class CodeCompletionTests
+{
+    private static void AssertCompletions(ImmutableArray<TextEdit> actual, params string[] expected) =>
+        Assert.True(actual.Select(x => x.Text).ToHashSet().SetEquals(expected));
+
+    private static ImmutableArray<CompletionItem> GetCompletions(SyntaxTree tree, SemanticModel model, SyntaxPosition cursor)
+    {
+        var service = new CompletionService();
+        service.AddProvider(new KeywordCompletionProvider());
+        service.AddProvider(new ExpressionCompletionProvider());
+        service.AddProvider(new MemberAccessCompletionProvider());
+        return service.GetCompletions(tree, model, cursor);
+    }
+
+    [Fact]
+    public void TestLocalCompletionGlobalVariable()
+    {
+        string code = """
+            val global = 5;
+            func main(){
+                var local = gl
+            }
+            """;
+        var tree = SyntaxTree.Parse(SourceText.FromText(code));
+        var cursor = new SyntaxPosition(2, 18);
```

Comment:
Roslyn tests use a pipe character and use that to determine the cursor position, making the tests a little more flexible. If you want, you could utilize that, `SourceText` can help you convert an index to syntax position